### ISSUE TITLE
FailedPreconditionError (see above for traceback): Attempting to use uninitialized value dense_93/bias

### ISSUE THREAD

**Issue by diemtran1005** [tensorflow/tensorflow#28427]
I run my code, but I don't know what is the below error. Please help me to fix this
FailedPreconditionError (see above for traceback): Attempting to use uninitialized value dense_93/bias 	 [[node dense_93/bias/read (defined at C:/Users/TRAN THI DIEM/Documents/diem/research/ReinforcementLearning/code/Depression/Code detecting depression using multi layer with survey data_ da chay duoc/depression_test_reinforcement.py:126) ]]

import tensorflow as tf
import pandas as pd
import numpy as np

from sklearn.utils import shuffle

from sklearn import preprocessing
from sklearn.model_selection import train_test_split
#train set
df_train = pd.read_csv('data/train.csv')
# test set
df_test = pd.read_csv('data/test.csv')

#show the shape of the train dataframe
df_train.shape
def missing_values_table(df):
        # Total missing values
        mis_val = df.isnull().sum()  # sum the number of each colum that has no value
        
        # Percentage of missing values
        mis_val_percent = 100 * df.isnull().sum() / len(df) #len(df); return number of row in df_train
        
        # Make a table with the results
        mis_val_table = pd.concat([mis_val, mis_val_percent], axis=1)
        
        # Rename the columns
        mis_val_table_ren_columns = mis_val_table.rename(
        columns = {0 : 'Missing Values', 1 : '% of Total Values'})
        
        # Sort the table by percentage of missing descending
        mis_val_table_ren_columns = mis_val_table_ren_columns[
            mis_val_table_ren_columns.iloc[:,1] != 0].sort_values(
        '% of Total Values', ascending=False).round(1)
        
        # Print some summary information
        print ("The dataset has " + str(df.shape[1]) + " columns.\n"      
            "There are " + str(mis_val_table_ren_columns.shape[0]) +
              " columns that have missing values.")
        
        # Return the dataframe with missing information
        return mis_val_table_ren_columns
    # Get the columns with > 50% missing
missing_df = missing_values_table(df_train);
missing_columns = list(missing_df[missing_df['% of Total Values'] > 50].index)
print('\n','%d columns will be deleted.' % len(missing_columns))

# Drop the columns with 50% missing data
df_train = df_train.drop(columns = list(missing_columns)) # delete some column have miss information more than 50%
# shape for df_train now is (1143,71)compare with the first (1143,75)

# Create a label (category) encoder object
#ncoder = LabelEncoder()
encoder = preprocessing.LabelEncoder()

# fitting the encoder to the "survey_date" column
encoder.fit(df_train['survey_date'])

# Apply the fitted encoder to the "survey_date" to transform categories into integers
encoded_train = encoder.transform(df_train['survey_date'])
# encoded_test = encoder.transform(df_test['survey_date'])

#assign the tranformed column back to the dataframe
df_train['survey_date'] = encoded_train
# split data into train and test sets
X = df_train.drop(["depressed"], axis=1)

# fill missing values with mean column values

imputer = preprocessing.Imputer()
transformed_X = imputer.fit_transform(X)

y_temp = df_train.depressed
y= df_train['depressed']
y_matrix = y.reset_index().values
y =y_matrix[:,1]

seed = 5
test_size = 0.33

X_train, X_test, y_train, y_test =train_test_split(transformed_X, y, test_size=test_size, random_state=seed)

input_size = [70]
class DQNetwork:
    def __init__(self):
#        self.state_size = state_size
 #       self.action_size = action_size
    
        self.learning_rate = tf.placeholder(tf.float32)
        #self.weight_decay = tf.constant(1e-4)

#        self.action = tf.placeholder(tf.uint16, [None,3])  
        # We create the placeholders
        # *state_size means that we take each elements of state_size in tuple hence is like if we wrote
        
        # [None, 84, 84, 4]
        self.inputs_ = tf.placeholder(tf.float64, [None,*input_size])
        
#        self.actions_ = tf.placeholder(tf.float32, [None, 3])

        # Remember that target_Q is the R(s,a) + ymax Qhat(s', a')
        self.label = tf.placeholder(tf.int64,[None])    
#fully connected        
        
           

       # self.flatten = tf.layers.flatten(self.conv6_Dropout)
        ## --> [43,264]

        self.output_layer1 = tf.layers.dense(inputs=self.inputs_,
                                     #    bias_initializer=tf.zeros_initializer(), kernel_initializer=tf.zeros_initializer(),
                                      #kernel_initializer=tf.contrib.layers.xavier_initializer_conv2d(),
                                     # kernel_regularizer=tf.contrib.layers.l2_regularizer(1e-4),
                                      units=36,
                                     activation=tf.nn.elu)
        self.output_layer2 = tf.layers.dense(inputs=self.output_layer1,
                                      #       bias_initializer=tf.zeros_initializer(), kernel_initializer=tf.zeros_initializer(),
                                      #kernel_initializer=tf.contrib.layers.xavier_initializer_conv2d(),
                                     # kernel_regularizer=tf.contrib.layers.l2_regularizer(1e-4),
                                      units=36,
                                    activation=tf.nn.elu)
        self.output = tf.layers.dense(inputs=self.output_layer2,
                                     #bias_initializer=tf.zeros_initializer(), kernel_initializer=tf.zeros_initializer(),
                                      #kernel_initializer=tf.contrib.layers.xavier_initializer_conv2d(),
                                     # kernel_regularizer=tf.contrib.layers.l2_regularizer(1e-4),
                                      units=1,
                                     activation=tf.nn.sigmoid)
        
        self.targets1 = tf.squeeze(tf.cast(self.label, tf.int32))   # Get predicted values by finding which logit is the greatest
        self.targets = tf.cast(tf.argmax(self.targets1, 1), tf.int32)        
        self.batch_predictions = tf.cast(tf.argmax(self.output, 1), tf.int32)
        self.predicted_correctly = tf.equal(self.batch_predictions, self.targets)
    # Average the 1's and 0's (True's and False's) across the batch size
        self.accuracy = tf.reduce_mean(tf.cast(self.predicted_correctly, tf.float32))
        
        
        
        self.loss = tf.reduce_mean( tf.nn.sparse_softmax_cross_entropy_with_logits(logits= self.output, labels=self.label) )
        #cross_entropy = tf.nn.sparse_softmax_cross_entropy_with_logits_v2(logits = self.output,label = self.label)
       # self.loss = tf.reduce_mean(cross_entropy, name='cross_entropy')
        self.optimizer = tf.train.AdamOptimizer(self.learning_rate).minimize(self.loss)
        #self.optimizer = tf.train.AdamOptimizer().minimize(self.loss)
# create model
#model = Sequential()
#model.add(Dense(36, input_dim=70, activation='elu'))
#model.add(Dense(36, activation='elu'))
#model.add(Dense(1, activation='sigmoid'))
## Compile model
#model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])
## Fit the model
#model.fit(X_train, y_train, epochs=100, batch_size=10)
#scores = model.evaluate(X_test, y_test)
#print("\n%s: %.2f%%" % (model.metrics_names[1], scores[1]*100))
learning_rate = 0.0001
sess = tf.Session()
sess.run(tf.global_variables_initializer())
dqn = DQNetwork ()
_,current_loss,current_accuracy = sess.run([dqn.optimizer,dqn.loss,dqn.accuracy],
                                                  feed_dict={dqn.inputs_:X_train,dqn.label:y_train,dqn.learning_rate:learning_rate})

**Comment by muddham** [comment]
@diemtran1005 It looks like you haven't used a template to create this issue. Please resubmit your issue using a template from [here](https://github.com/tensorflow/tensorflow/issues/new/choose). We ask users to use the template because it reduces overall time to resolve a new issue by avoiding extra communication to get to the root of the issue. We will close this issue in lieu of the new one you will create from the template. Thank you for your cooperation.

**Comment by muddham** [comment]
@diemtran1005 Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
